views/modalTemplate.pug

```diff
@@ -0,0 +1,30 @@
+mixin modalTemplate(action, title, desc, buttonText, isActionFunction)
+  div(class="modal hidden fixed z-10 inset-0 overflow-y-auto" aria-labelledby="modal-title" role="dialog" aria-modal="true")
+    div(class="flex items-end justify-center min-h-screen pt-4 px-4 pb-20 text-center sm:block sm:p-0")
+      //- modal overlay
+      div(class="fixed inset-0 bg-gray-500 bg-opacity-75 transition-opacity" onClick='toggleModal()' aria-hidden="true")
+
+      span(class="hidden sm:inline-block sm:align-middle sm:h-screen" aria-hidden="true") &#8203;
+      //- modal
+      div(class="inline-block align-bottom bg-white rounded-lg text-left overflow-hidden shadow-xl transform transition-all sm:my-8 sm:align-middle sm:max-w-lg sm:w-full")
+        div(class="bg-white px-4 pt-5 pb-4 sm:p-6 sm:pb-4")
+          div(class="sm:flex sm:items-start")
+            div(class="mx-auto flex-shrink-0 flex items-center justify-center h-12 w-12 rounded-full bg-red-100 sm:mx-0 sm:h-10 sm:w-10")
+              //-- Heroicon name: outline/exclamation
+              svg(class="h-6 w-6 text-red-400" xmlns="http://www.w3.org/2000/svg" fill="none" viewBox="0 0 24 24" stroke="currentColor" aria-hidden="true")
+                path(stroke-linecap="round" stroke-linejoin="round" stroke-width="2" d="M12 9v2m0 4h.01m-6.938 4h13.856c1.54 0 2.502-1.667 1.732-3L13.732 4c-.77-1.333-2.694-1.333-3.464 0L3.34 16c-.77 1.333.192 3 1.732 3z" )
+            div(class="mt-3 text-center sm:mt-0 sm:ml-4 sm:text-left")
+              h3(class="text-lg leading-6 font-medium text-gray-900" id="modal-title")= title
+              div(class="mt-2")
+                p(class="text-sm text-gray-500" id='modal-desc')= desc
+        div(class="bg-gray-50 px-4 py-3 sm:px-6 sm:flex sm:flex-row-reverse")
+          form(action=action, method="post" id='modal-form' class=`${isActionFunction ? 'hidden':''}`)
+            button(type="submit" id='submit-button' class="w-full inline-flex justify-center rounded-md border border-transparent shadow-sm px-4 py-2 btn btn-primary text-base font-medium text-white animate-hover focus:outline-none focus:ring-2 focus:ring-offset-2 focus:ring-red-400 sm:ml-3 sm:w-auto sm:text-sm")= buttonText
+          button(type="button" id='func-button' onClick=`toggleModal();${action};` class=`${!isActionFunction ? 'hidden' : ''} w-full inline-flex justify-center rounded-md border border-transparent shadow-sm px-4 py-2 btn btn-primary text-base font-medium text-white animate-hover focus:outline-none focus:ring-2 focus:ring-offset-2 focus:ring-red-400 sm:ml-3 sm:w-auto sm:text-sm`)= buttonText
+          button(type="button" onClick='toggleModal()' class="mt-3 w-full inline-flex justify-center rounded-md border border-gray-300 shadow-sm px-4 py-2 bg-white text-base font-medium text-gray-700 hover:bg-gray-100 focus:outline-none focus:ring-2 focus:ring-offset-2 focus:ring-gray-100 sm:mt-0 sm:ml-3 sm:w-auto sm:text-sm") Mégse
```

Comment:
Since all the buttons on the site use a pill-shaped design, I would stick with that on the modal too.